deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorStreamCodeGen.java

```diff
@@ -0,0 +1,104 @@
+package io.quarkiverse.openapi.generator.deployment.codegen;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.zip.CRC32;
+import java.util.zip.CheckedInputStream;
+
+import org.eclipse.microprofile.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkiverse.openapi.generator.codegen.OpenApiSpecInputProvider;
+import io.quarkiverse.openapi.generator.codegen.SpecInputModel;
+import io.quarkus.bootstrap.prebuild.CodeGenException;
+import io.quarkus.deployment.CodeGenContext;
+
+public class OpenApiGeneratorStreamCodeGen extends OpenApiGeneratorCodeGenBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenApiGeneratorStreamCodeGen.class);
+
+    private List<OpenApiSpecInputProvider> providers;
+
+    public OpenApiGeneratorStreamCodeGen() {
+
+    }
+
+    private void loadServices() {
+        final ServiceLoader<OpenApiSpecInputProvider> loader = ServiceLoader.load(OpenApiSpecInputProvider.class);
+        providers = loader.stream().map(ServiceLoader.Provider::get).collect(Collectors.toList());
+    }
+
+    @Override
+    public String providerId() {
+        return "open-api-stream";
+    }
+
+    // unused by this CodeGenProvider since we rely on the input coming from ServiceLoaders
+    @Override
+    public String inputExtension() {
+        return ".yaml";
+    }
+
+    // unused by this CodeGenProvider since we rely on the input coming from ServiceLoaders
+    @Override
+    public String inputDirectory() {
+        return "openapi";
+    }
+
+    @Override
+    public boolean trigger(CodeGenContext context) throws CodeGenException {
+        final Path outDir = context.outDir();
+        boolean generated = false;
+
+        for (final OpenApiSpecInputProvider provider : this.providers) {
+            for (SpecInputModel inputModel : provider.read()) {
+                if (inputModel == null) {
+                    throw new CodeGenException("SpecInputModel from provider " + provider + " is null");
+                }
```

Comment:
Im not sure if this is the proper flow
I mean, you are reading the input stream and copying it to a local file, is that really needed? 
Does not the generator directly accepts the input stream to generate the content?